dandischema/models.py

```diff
@@ -1116,19 +1115,26 @@ class BareAsset(CommonModel):
     }
 
     @validator("digest")
-    def digest_etag(cls, values):
-        try:
-            digest = values[DigestType.dandi_etag]
-            if "-" not in digest or len(digest.split("-")[0]) != 32:
-                raise ValueError
-        except KeyError:
-            raise ValueError("Digest is missing dandi-etag value.")
-        except ValueError:
-            raise ValueError(
-                f"Digest must have an appropriate dandi-etag value. "
-                f"Got {values[DigestType.dandi_etag]}"
-            )
-        return values
+    def digest_check(cls, values):
```

Comment:
If you rename `values` to something else and add a separate `values` argument for receiving previously-validated fields, you'll be able to take `encodingFormat` into consideration, thereby letting the code determine precisely which of `dandi_etag` and `dandi_zarr_checksum` is required.

Reply:
No, it doesn't need to be a root validator.  See the `password2` validator [in this page of the pydantic docs](https://pydantic-docs.helpmanual.io/usage/validators/) for an example.

Reply:
thanks. there are a few changes that has been made based on this which simplifies the checking.